server/src/main/java/org/elasticsearch/search/vectors/KnnVectorQueryBuilder.java

```diff
@@ -28,74 +28,102 @@
 import org.elasticsearch.index.query.QueryBuilder;
 import org.elasticsearch.index.query.QueryRewriteContext;
 import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
 import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
 /**
  * A query that performs kNN search using Lucene's {@link org.apache.lucene.search.KnnFloatVectorQuery} or
  * {@link org.apache.lucene.search.KnnByteVectorQuery}.
  *
- * NOTE: this is an internal class and should not be used outside of core Elasticsearch code.
  */
 public class KnnVectorQueryBuilder extends AbstractQueryBuilder<KnnVectorQueryBuilder> {
     public static final String NAME = "knn";
+    private static final int NUM_CANDS_LIMIT = 10000;
+    public static final ParseField FIELD_FIELD = new ParseField("field");
+    public static final ParseField NUM_CANDS_FIELD = new ParseField("num_candidates");
+    public static final ParseField QUERY_VECTOR_FIELD = new ParseField("query_vector");
+    public static final ParseField VECTOR_SIMILARITY_FIELD = new ParseField("similarity");
```

Comment:
I am leaning towards just a `vector_similarity` query, not a parent query, etc. and  not using this parameter when used as a `query`.

Reply:
If we want to remove `similarity` field, we need to introduce another QueryBuilder separate from the current  `KnnVectorQueryBuilder`that is used by knn search and uses `similarity` field.

Reply:
@mayya-sharipova I am ok having it their for now. It is simplest and keeps uniformity with top-level knn. 

If we ever introduce a `vector_similarity_query`, we would deprecate this parameter and point folks to use it instead.